src/h5web/visualizations/shared/utils.ts

```diff
@@ -48,40 +55,48 @@ export function findDomain(data: number[]): Domain | undefined {
     : undefined;
 }
 
+export function isIndexAxisConfig(
+  config: AxisConfig
+): config is IndexAxisConfig {
+  return 'indexDomain' in config;
+}
+
 export function getDataScaleFn(isLog: boolean): DataScaleFn {
   return isLog ? scaleSymlog : scaleLinear;
 }
 
-export function useAbscissaScale(): {
-  abscissaScale: DataScale;
-  abscissaScaleFn: DataScaleFn;
-} {
-  const { size } = useThree();
-  const { width } = size;
-  const { axisDomains, hasXLogScale } = useContext(AxisSystemContext);
+function useAxisScale(config: AxisConfig, rangeSize: number): AxisScale {
+  if (isIndexAxisConfig(config)) {
+    const { indexDomain } = config;
+    const scaleFn = scaleLinear;
+
+    const scale = scaleFn();
+    scale.domain(indexDomain);
+    scale.range([-rangeSize / 2, rangeSize / 2]);
+
+    return { scale, scaleFn, domain: indexDomain };
+  }
+
+  const { dataDomain, isLog = false } = config;
+  const scaleFn = getDataScaleFn(isLog);
 
-  const abscissaScaleFn = getDataScaleFn(!!hasXLogScale);
-  const abscissaScale = abscissaScaleFn();
-  abscissaScale.domain(axisDomains.x);
-  abscissaScale.range([-width / 2, width / 2]);
+  const scale = scaleFn();
+  scale.domain(dataDomain);
+  scale.range([-rangeSize / 2, rangeSize / 2]);
```

Comment:
The scale creation code looks a bit duplicated for index/data axis, but I kept it separate because it will probably diverge in the future, when we add `centerTicks`, notably.